vignettes/KEGGgraph.Rmd

```diff
@@ -0,0 +1,516 @@
+if(require(org.Hs.eg.db)) {
+  ioGeneID <- translateKEGGID2GeneID(names(ios))
+  nodesNames <- sapply(mget(ioGeneID, org.Hs.egSYMBOL, ifnotfound=NA), "[[",1)
+} else {
+  nodesNames <- names(ios)
+}
+names(nodesNames) <- names(ios)
+
+nAttrs <- list();
+nAttrs$fillcolor <- makeAttr(mapkGsub, "lightgrey", list(orange=names(ios)[ios]))
+nAttrs$label <- nodesNames
+plot(mapkGsub, "neato", nodeAttrs=nAttrs,
+     attrs=list(node=list(fillcolor="lightgreen",
+                  width="0.75", shape="ellipse"), 
+       edge=list(arrowsize="0.7")))
+``` 
+
+Another common operation on graphs is merging, that is, combining different
+graphs together. It is inspired by the fact that many KEGG pathways embed other
+pathway, for example MAPK signaling pathway embeds 6 pathways including Wnt
+signaling pathway. `mergeGraphs` provides the possibility to merge them into one
+graph for further analysis. Next we merge MAPK and Wnt signaling pathway into
+one graph. The graphs to be merged should be organized into a list, and it is
+commandary to use 'expandGenes=TRUE' option when parsing to make sure the nodes
+are unique and indexed by KEGGID.
+
+```{r mergedemo, results='verbatim'}
+wntKGML <- system.file("extdata/hsa04310.xml",package="KEGGgraph")
+wntG <- parseKGML2Graph(wntKGML)
+graphs <- list(mapk=mapkG, wnt=wntG)
+merged <- mergeGraphs(graphs)
+merged
+``` 
+
+We observe that the node number in the merged graph (`r numNodes(merged)`) is
+less than the sum of two graphs (`r numNodes(mapkG)` and `r numNodes(wntG)` for
+MAPK and Wnt pathway respectively), reflecting the crosstalk between the
+pathways by sharing nodes.
+
+## Using other graph tools
+
+In **R** and **Bioconductor** there'are powerful tools for graph algorithms and
+operations, including `r Biocpkg("graph")`, `r Biocpkg("Rgraphviz")` and 
+`r Biocpkg("RBGL")`. The KEGG graphs can be analyzed with their functionalities to
+describe characterisitcs of the pathway and to answer biological relevant
+questions.
+
+Here we demonstrate the use of other graph tools with asking the question which
+nodes are of the highest importance in MAPK signalling pathway. To this end we
+turn to relative betweenness centrality [@Aittokallio06], [@Carey05]. Betweenness
+is a centrality measure of a node within a graph. Nodes that occur on many
+shortest paths between other vertices have higher betweenness than those that do
+not. It is scaled by the factor of (n-1)(n-2)/2 to get relative betweenness
+centrality, where n is the number of nodes in the graph. Both measurements
+estimate the importance or the role of the node in the graph.
+
+With the function implemented in `r Biocpkg("RBGL")`, our aim is to identify
+most important nodes in MAPK signalling pathway.
+
+*Note*: the current version of `RBGL` (version 1.59.5) reports the error  that
+`BGL_brandes_betweeness_centrality` not available for `.Call()` for package 
+`r Biocpkg("RBGL")`. Therefore the execution has been suppressed for now.
```

Comment:
@Accio I just wanted to make a note of this section that references RBGL 1.59.5 (it's now 1.79.0).